app/models/user.rb

```diff
@@ -253,7 +259,20 @@ def mobile_verification_missing?
   end
 
   def image_url
-    identity&.image_url&.gsub("http://", "//")
+    identity&.image_url&.gsub("http://", "//") || gravatar_image_url
+  end
+
+  def gravatar_image_url
+    hash = Digest::MD5.hexdigest(email.downcase)
+    return "https://secure.gravatar.com/avatar/#{hash}?d=identicon&s=80"
+  end
+
+  def social_profile?
+    identity.present?
+  end
```

Comment:
Identity currently has a provider type of "identity", which I assume means there is a user / email, but no social profile. If so, this method feels like it should return identity.facebook || identity.twitter.

Reply:
I think for email it returns a `mailto:` ink, see `profile_url` below?